dnf/base.py

```diff
@@ -212,6 +228,100 @@ def _setup_excludes_includes(self, only_main=False):
+        for repo in self.repos.iter_enabled():
+            try:
+                module_metadata = ModuleMetadataLoader(repo).load()
+
+                defaults = [i for i in module_metadata if isinstance(i, Modulemd.Defaults)]
+                module_defaults_prioritizer.add(defaults, 0)
+
+                # read modules from modulemd
+                for data in module_metadata:
+                    if not isinstance(data, Modulemd.Module):
+                        continue
+                    self.repo_module_dict.add(RepoModuleVersion(data, base=self, repo=repo))
+            except (dnf.exceptions.Error, GLib.Error) as e:
+                logger.debug(e)
+                continue
+
+        # read defaults from disk
+        # chroot behavior:
+        #   * use host configuration (repos work the same)
+        #   * to use chroot configuration, you need to switch to chroot
+        topdir = self.conf.moduledefaultsdir._get()
+        try:
+            for fn in os.listdir(topdir):
+                if not fn.endswith(".yaml"):
+                    continue
+                yaml_file = os.path.join(topdir, fn)
+
+                with open(yaml_file, "r") as f:
+                    modules_yaml = f.read()
+                module_metadata = Modulemd.Module.new_all_from_string_ext(modules_yaml)
+                defaults = [i for i in module_metadata if isinstance(i, Modulemd.Defaults)]
+                module_defaults_prioritizer.add(defaults, 1000)
+        except OSError as ex:
+            if ex.errno != errno.ENOENT:
+                raise
+
+        # resolve defaults and store them for future use
+        for default in module_defaults_prioritizer.resolve():
+            try:
+                self.repo_module_dict[default.peek_module_name()].defaults = default
+            except KeyError:
+                logger.debug("No module named {}, skipping.".format(default.peek_module_name()))
+
+        self.repo_module_dict.read_all_module_confs()
+        self._module_persistor = ModulePersistor()
+
+    def use_module_includes(self):
+        def update_include_nevras(name, stream):
+            include_set, _ = self.repo_module_dict.get_includes(name, stream)
+            include_nevras_set.update(include_set)
+
+        include_nevras_set = set()
+        exclude_nevras_set = set()
+
+        for repo_module in self.repo_module_dict.values():
+            if repo_module.conf.enabled._get():
+                update_include_nevras(repo_module.name, repo_module.conf.stream._get())
+            elif repo_module.defaults.peek_default_stream():
+                update_include_nevras(repo_module.name, repo_module.defaults.peek_default_stream())
+
+            exclude_set, _ = self.repo_module_dict.get_excludes(repo_module.name)
+            exclude_nevras_set.update(exclude_set)
+
+        # collect all hotfix repo repoids - we don't filter them at all
+        hotfix_repos = [i.id for i in self.repos.iter_enabled() if i.hotfixes._get()]
```

Comment:
Please can you add following repos to black list:
@commandline
@system
as
hawkey.SYSTEM_REPO_NAME
hawkey.CMDLINE_REPO_NAME

Reply:
Fixed.